python/nav/ipdevpoll/neighbor.py

```diff
@@ -112,20 +111,20 @@ def __init__(self, record, local_address=None):
         if local_address:
             self._invalid_neighbor_ips.append(str(local_address))
 
-        self.netbox = self.interface = None
```

Comment:
One real world scenario here is where switch A sees 3 different logical interfaces from switch B, and switch B sees 1 interface of switch A on one of it's 3 logical interfaces. The old code did deal with this situation by picking an arbitrary one and discarding the rest in the collection phase. I changed this to make the selection in the topology algorithm where we can actually find the proper reverse link. I think this is also a cleaner way to do it from a design perspective. Now the data collection plugin collects data and the topology service detects topology. Whereas before the data collection plugin would do part of the topology detection on missing data